Documentation/contributing/development/debugging.rst

```diff
@@ -4,10 +4,69 @@
     Please use the official rendered version released here:
     https://docs.cilium.io
 
+.. _gs_debugging:
+
 #########
 Debugging
 #########
 
+Attaching a debugger
+--------------------
+
+Cilium comes with a handy set of Makefile targets for quickly deploying
+development builds to a local :ref:`Kind <gs_kind>` cluster. One of them is
+``kind-debug-agent``, generating a container image that wraps the Cilium agent
+with a `dlv <https://github.com/go-delve/delve>`_ invocation, causing the agent
+process to listen for connections from a debugger frontend on port 2345.
+
+To build and push a debug image to your local Kind cluster, run:
+
+.. code-block:: shell-session
+
+    $ make kind-debug-agent
+
+.. note::
+      The image is automatically pushed to the Kind nodes, but running Cilium
+      Pods are not restarted.
+
+If your Kind cluster was set up using ``make kind``, it will automatically
+be configured using with the following port mappings:
+
+- ``23401``: First ``kind-control-plane`` node
+- ``2340*``: Subsequent ``kind-control-plane`` nodes, if defined
+- ``23411``: First ``kind-worker`` node
+- ``2341*``: Subsequent ``kind-worker`` nodes, if defined
+
+The Delve listener supports multiple debugging protocols, so any IDEs or
+debugger frontends that understand either DAP or Delve API v2 are supported.
+
+~~~~~~~~~~~~~~~~~~
+Visucl Studio Code
+~~~~~~~~~~~~~~~~~~
+
+The Cilium repository contains a VS Code launch configuration
+(``.vscode/launch.json``) with the most common debug targets for Kind: the
+control plane, the first two ``kind-worker`` nodes and the :ref:`Cilium Operator
+<cilium_operator_internals>`.
+
+.. image:: _static/vscode-run-and-debug.png
+    :align: center
+|
+
+The screenshot above is taken from the 'Run And Debug' section (``Shift+Ctrl+D``
+by default). Select a target to attach to, start the debug session and set a
```

Comment:
There are a couple of changes that we can make here to make this more clear for users:

> The preceding screenshot is taken from the 'Run And Debug' section in Visual Studio Code. The default shortcut to access this section is ``Shift+Ctrl+D``.

* Preceding is preferred over Above, as it allows for easier language localization.
* We can remove the parentheses to make this more direct.